posthog/tasks/warehouse.py

```diff
@@ -0,0 +1,203 @@
+            team_id=resource.team.pk,
+            access_key=settings.AIRBYTE_BUCKET_KEY,
+            access_secret=settings.AIRBYTE_BUCKET_SECRET,
+        )
+
+        data = {
+            "credential": credential,
+            "name": "stripe_customers",
+            "format": "Parquet",
+            "url_pattern": f"https://{settings.AIRBYTE_BUCKET_DOMAIN}/airbyte/{resource.team.pk}/customers/*.parquet",
+            "team_id": resource.team.pk,
+        }
+
+        table = DataWarehouseTable(**data)
+        try:
+            table.columns = table.get_columns()
+        except Exception as e:
+            logger.exception(
+                f"Data Warehouse: Sync Resource failed with an unexpected exception for connection: {resource.connection_id}",
+                exc_info=e,
+            )
+        else:
+            table.save()
+
+            resource.are_tables_created = True
+            resource.status = job["status"]
+            resource.save()
+
+    else:
+        resource.status = job["status"]
+        resource.save()
+
+
+DEFAULT_USAGE_LIMIT = 1000000
+ROWS_PER_DOLLAR = 66666  # 1 million rows per $15
+
+
+@app.task(ignore_result=True, max_retries=2)
+def check_external_data_source_billing_limit_by_team(team_id):
+    from posthog.warehouse.external_data_source.connection import deactivate_connection_by_id, activate_connection_by_id
+
+    team = Team.objects.get(pk=team_id)
+    all_active_connections = ExternalDataSource.objects.filter(team=team, status__in=["running", "succeeded"])
+    all_inactive_connections = ExternalDataSource.objects.filter(team=team, status="inactive")
+
+    _usage_limit = _get_data_warehouse_usage_limit(team_id)
+
+    # TODO: consider more boundaries
+    if _usage_limit and team.external_data_workspace_rows_synced_in_month >= (_usage_limit * ROWS_PER_DOLLAR):
+        for connection in all_active_connections:
+            deactivate_connection_by_id(connection.connection_id)
+            connection.status = "inactive"
+            connection.save()
+    else:
+        for connection in all_inactive_connections:
+            activate_connection_by_id(connection.connection_id)
+            connection.status = "running"
+            connection.save()
+
+
+def _get_data_warehouse_usage_limit(team_id):
+    from posthog.cloud_utils import get_cached_instance_license
+    from ee.billing.billing_manager import BillingManager
+
+    license = get_cached_instance_license()
+    team = Team.objects.get(pk=team_id)
+    org = team.organization
+
+    response = BillingManager(license).get_billing(org, None)
+    try:
+        data_warehouse_product = [product for product in response["products"] if product["name"] == "Data Warehouse"][0]
+        usage_limit = data_warehouse_product["usage_limit"]
+    except Exception as e:
+        logger.exception("Data Warehouse: Failed to retrieve data warehouse billing product", exc_info=e)
+        usage_limit = DEFAULT_USAGE_LIMIT
```

Comment:
It feels like all this should be in `quota_limiting.py` - any reason you didn't include it there / use the existing methods but extend for data warehouse?

Reply:
Ah i see. Makes sense. Wasn't aware of this file